ucum-java8/src/test/java/systems/uom/ucum/format/UnitFormatTest.java

```diff
@@ -66,181 +51,353 @@
+
+		assertEquals("m/s", format.format(v));
 	}
 
 	@Test
 	public void testFormatUCUMCS() {
 		final UnitFormat format = UCUMFormat.getInstance(CASE_SENSITIVE);
-		final Appendable a = new StringBuilder();
-		try {
-			format.format(METRE, a);
-		} catch (IOException e) {
-			fail(e.getMessage());
-		}
-		assertEquals(METRE, sut.getUnit());
-		assertEquals("m", a.toString());
-
-		final Appendable a2 = new StringBuilder();
-		@SuppressWarnings("unchecked")
-		Unit<Speed> v = (Unit<Speed>) METER.divide(SECOND);
-		try {
-			format.format(v, a2);
-		} catch (IOException e) {
-			fail(e.getMessage());
-		}
-		assertEquals("m/s", a2.toString());
-	}
-	
-	@Test
-	public void testFormatUCUMCSfromQuantity() {
-		final UnitFormat format = UCUMFormat.getInstance(CASE_SENSITIVE);
-		final Appendable a = new StringBuilder();
-		try {
-			format.format(METRE, a);
-		} catch (IOException e) {
-			fail(e.getMessage());
-		}
-		assertEquals(METRE, sut.getUnit());
-		assertEquals("m", a.toString());
-
-		final Appendable a2 = new StringBuilder();
-		@SuppressWarnings("unchecked")
-		Unit<Speed> v = (Unit<Speed>) sut.getUnit().divide(SECOND);
-		try {
-			format.format(v, a2);
-		} catch (IOException e) {
-			fail(e.getMessage());
-		}
-		assertEquals("m/s", a2.toString());
+
+		assertEquals(METER, sut.getUnit());
+		assertEquals("m", format.format(METER));
+
+		Unit<Speed> v = new ProductUnit<Speed>(METER.divide(SECOND));
+
+		assertEquals("m/s", format.format(v));
 	}
 
 	@Test
 	public void testFormatUCUMCI() {
 		final UnitFormat format = UCUMFormat.getInstance(CASE_INSENSITIVE);
-		final Appendable a = new StringBuilder();
-		try {
-			format.format(METRE, a);
-		} catch (IOException e) {
-			fail(e.getMessage());
-		}
-		assertEquals(METRE, sut.getUnit());
-		assertEquals("M", a.toString());
+
+		assertEquals(METER, sut.getUnit());
+		assertEquals("M", format.format(METER));
 	}
 
 	@Test(expected=UnsupportedOperationException.class)
 	public void testParseLocal() {
 		final UnitFormat format = LocalUnitFormat.getInstance();
-		try {
-			Unit<?> u = format.parse("min");
-			assertEquals("min", u.getSymbol());
-		} catch (ParserException e) {
-			fail(e.getMessage());
-		}
+
+		assertEquals("min", format.parse("min").getSymbol());
 	}
 
 	@Test
 	public void testParseUCUMCS() {
 		final UnitFormat format = UCUMFormat.getInstance(CASE_SENSITIVE);
-		try {
-			Unit<?> u = format.parse("min");
-			assertEquals(MINUTE, u);
-		} catch (ParserException e) {
-			fail(e.getMessage());
-		}
+
+		assertEquals(MINUTE, format.parse("min"));
 	}
 
+    @Test(expected=TokenException.class)
+    public void testParseUCUMCSError() {
+        final UnitFormat format = UCUMFormat.getInstance(CASE_SENSITIVE);
+
+        assertEquals(MINUTE, format.parse("MIN"));
+    }
+
+    @Test
+    public void testParseUCUMCSLowercaseLiter() {
+        final UnitFormat format = UCUMFormat.getInstance(CASE_SENSITIVE);
+
+        assertEquals(LITER, format.parse("l"));
```

Comment:
See this is the sort of thing you'll have to reconcile with #82. In case-sensitive mode, are `"l"` and `"L"` really the same units? (I would help they are, but I'm still scared because of what @keilw had said about them being separate units in the UCUM world; see #82 for more information.) If not, then one of these assertions should fail.